dhctl/pkg/kubernetes/actions/converge/state.go

```diff
@@ -101,11 +101,14 @@ func GetClusterStateFromCluster(kubeCl *client.KubernetesClient) ([]byte, error)
 
 // Create secret for node with group settings only.
 func CreateNodeTerraformState(kubeCl *client.KubernetesClient, nodeName, nodeGroup string, settings []byte) error {
+	secret, err := manifests.SecretWithNodeTerraformState(nodeName, nodeGroup, nil, settings)
+	if err != nil {
+		return err
+	}
+
 	task := actions.ManifestTask{
-		Name: fmt.Sprintf(`Secret "d8-node-terraform-state-%s"`, nodeName),
-		Manifest: func() interface{} {
-			return manifests.SecretWithNodeTerraformState(nodeName, nodeGroup, nil, settings)
-		},
+		Name:     fmt.Sprintf(`Secret "d8-node-terraform-state-%s"`, nodeName),
```

Comment:
Can we migrate to hashes to solve the problem once and forever? 
Firstly check the secret with the hash of the node name and it there is not found error, try to search with the actual node name.

Or, if we know the allowed length of the `nodeName`, we can add the validation to OpenAPI spec.